content/components/selectpanel.mdx

```diff
@@ -0,0 +1,222 @@
+    />
+    <Caption>
+      Pressing esc, or selecting “cancel” should close the current dialog and return focus to the previous trigger. It
+      should not retain or submit any user input.
+    </Caption>
+  </div>
+  <div>
+    <CustomVideoPlayer
+      width="100%"
+      src="https://user-images.githubusercontent.com/40274682/228390297-d6d59818-1d1e-47bb-ac9d-cd1eccc57ba8.mp4"
+    />
+    <Caption>Selecting the submit button should submit any user input.</Caption>
+  </div>
+</Box>
+
+### Responsive
+
+Select panel can become a full screen dialog or a bottom sheet at narrow viewports. Support at 320px is required if the Select panel is larger at wider viewports. See [narrow-viewport placement](https://primer.style/design/components/dialog#narrow-viewport-placement) in the Dialog guidelines for more details.
+
+## Keyboard navigation
+
+### Trigger
+
+| Key                                                                                                                                                       | Description            |
+| --------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------- |
+| <Box as="kbd" backgroundColor="canvas.subtle" borderRadius={1} borderColor="border.default" borderWidth={1} borderStyle="solid" py={1} px={2}>Enter</Box> | Opens the Select panel. |
+| <Box as="kbd" backgroundColor="canvas.subtle" borderRadius={1} borderColor="border.default" borderWidth={1} borderStyle="solid" py={1} px={2}>Space</Box> | Opens the Select panel. |
+
+### Select panel
+
+| Key                                                                                                                                                                                          | Description                                                                                                                 |
+| -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
+| <Box as="kbd" backgroundColor="canvas.subtle" borderRadius={1} borderColor="border.default" borderWidth={1} borderStyle="solid" py={1} px={2}>Enter</Box>                                    | Submits the information in the form when focused in the listbox. Activates buttons or links focused within the Select panel. |
+| <Box as="kbd" backgroundColor="canvas.subtle" borderRadius={1} borderColor="border.default" borderWidth={1} borderStyle="solid" py={1} px={2}>Tab</Box>                                      | Move focus forward between interactive controls and widgets within the dialog.                                              |
```

Comment:
When the last control is focussed, does the `Tab` key cycle to the top of the controls within the select panel? In other words - are we focus trapping?

If we are focus trapping, does that give users who cannot see the page the same experience as those who can? If a sighted user is able to see the rest of the contents of the page, but a user who is reliant on moving focus is trapped within the bounds of the SelectPanel, that feels inconsistent to me.

Reply:
Yes, we need to be trapping focus. We need a solid boundary for these complex dialogs or else screen reader users will not know they've left the dialog and they will have to use context clues to realize they've left it. It could be a big problem if they perform a command thinking that they're still within the bounds of the dialog but they are not.